util/volume.go

```diff
@@ -0,0 +1,71 @@
+package util
+
+import (
+	"encoding/json"
+	"github.com/hpe-storage/common-host-libs/logger"
+	"github.com/hpe-storage/common-host-libs/model"
+)
+
+func GetVolumeObject(serialNumber, lunID string) *model.Volume {
+
+	var volObj *model.Volume
+	volObj.SerialNumber = serialNumber
+	volObj.LunID = lunID
+
+	return volObj
+}
+
+func GetSecondaryArrayLUNIds(details string) []int32 {
+	var secondaryArrayDetails model.SecondaryBackendDetails
+	logger.Tracef("\n About to unmarshal %s", details)
+	err := json.Unmarshal([]byte(details), &secondaryArrayDetails)
+	if err != nil {
+		logger.Tracef("\n Error in GetSecondaryArrayLUNIds %s", err.Error())
+		logger.Tracef("\n Passed string: %s", details)
+		return []int32{}
+	}
+	numberOfSecondaryBackends := len(secondaryArrayDetails.PeerArrayDetails)
+	var secondaryLunIds []int32 = make([]int32, numberOfSecondaryBackends)
+	for i := 0; i < numberOfSecondaryBackends; i++ {
+		secondaryLunIds[i] = secondaryArrayDetails.PeerArrayDetails[i].LunID
+	}
+	return secondaryLunIds
```

Comment:
Won't this return all secondary LUN IDs on all secondary arrays?  I thought we wanted to handle each secondaryArray independent of any other secondaryArray.  Isn't that more correct?

Reply:
Yes it will return all the secondary targets lun ids. While do a rescanFc/IscsiTarget() what we need is the list of lun id to iterate and issue "echo - - <lun-id>" >  /sys/... " so processing each lun id from individual secondary target or clubbing together all the lun id's (as done here) , the rescan logic is done for all luns together. what i have done is the seggregation in the logic for the Primary lun scan and all the luns for individual secondary targets.

Reply:
Will let @shivamerla comment here.  Our approach was to have a high level function to iterate over the secondary array details and perform the login + scan logic on each of them.  Clubbing all of these IDs together seems to defeat the purpose of having stored them in this fashion.

Reply:
@wdurairaj lets discuss this over call. I think what we want to achieve is a way to validate if all secondary luns are discovered and claimed by multipath. Lets see if we can add an additional call which does secondaryLunAttach() and perform discovery/scan on each. So, we don't touch existing logic anywhere and handle secondary ones separately.

Reply:
@rgcostea , @shivamerla , i have done some refactoring around handling secondary arrays, however the handleRemapLun(), handleOrphanPaths() i have kept the code intact, since i could'nt meaningfully split that code across primary and secondary backends(s) since once the volume's path are discovered, we need to handle the paths (based on lunids) as a single group only.

Let me know, if the change is looking ok or any other changes required.
I fixed the other review comments from @rgcostea  as well.

I have done testing around CSP response with/without secondary backends